### PR TITLE
Add Qwen3.5-35B-A3B MoE VLM ONNX export recipe

### PR DESCRIPTION
**Pull request by tanzeel-amd**
- Olive recipe for exporting Qwen/Qwen3.5-35B-A3B (256 experts, 8 routed + 1 shared)
- Three sub-model pipeline: text decoder (INT4 QMoE), embedding (FP32), vision (FP32)
- Custom ONNX-export-friendly MoE model class (codes/modeling_qwen3_5_moe.py)
- Inference script with text, image, interactive, and benchmark modes
- Requires ORT GenAI built with qwen3_5_moe support

### CODE REVIEW

**Comment by tanzeel-amd** [comment]
@microsoft-github-policy-service agree company="AMD"

**Comment by VishalX** [comment]
@xieofxie / @devang-ml pls review

**Comment by xieofxie** [comment]
please wait for https://github.com/microsoft/onnxruntime-genai/pull/2146

File: Qwen-Qwen3.5-35B-A3B/builtin/optimize.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+# --------------------------------------------------------------------------
+"""End-to-end optimization pipeline for Qwen3.5-35B-A3B MoE VLM.
+
+Exports three sub-models (vision encoder, text embedding, text decoder),
+applies graph optimizations and INT4 quantization via Olive passes.
+
+Usage:
+    python optimize.py --config-dir cpu_and_mobile --device cpu
+    python optimize.py --config-dir cpu_and_mobile --device cpu --skip-export
+"""
```
[the remark this answers is not in the document]
Comment (tanzeel-amd) [reply Qwen-Qwen3.5-35B-A3B/builtin/optimize.py:15]: Removed DEBUG_STATUS.md reference from description.

[the remark this answers is not in the document]
Comment (tanzeel-amd) [reply Qwen-Qwen3.5-35B-A3B/builtin/optimize.py:15]: Removed the unintended reference to DEBUG_STATUS in description.

File: Qwen-Qwen3.5-35B-A3B/builtin/user_script.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+    state_dict = {}
+    for sf in st_files:
+        tensors = load_file(sf)
+        for k, v in tensors.items():
+            if k.startswith("model."):
+                stripped = k[6:]
+                state_dict[stripped] = v
+                if stripped.startswith("language_model.embed_tokens."):
+                    state_dict[stripped[len("language_model."):]] = v
```
[the remark this answers is not in the document]
Comment (tanzeel-amd) [reply Qwen-Qwen3.5-35B-A3B/builtin/user_script.py]: Good point. Fixed in 7e3c0fa -- _load_model() now uses safetensors.safe_open with prefix filtering (model.visual.* and model.language_model.embed_tokens.*) to load only the vision encoder and embedding weights (~4 GB) instead of the full 35B checkpoint (~67 GB). Shards with no matching keys are skipped entirely without loading any tensors.

File: Qwen-Qwen3.5-35B-A3B/builtin/codes/modeling_qwen3_5_moe.py [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+#
+# Self-contained ONNX-export-friendly Qwen3.5-MoE model.
+# Vision encoder is identical to non-MoE Qwen3.5. MoE text components are
+# included for completeness but the text decoder is exported via OGA ModelBuilder.
+# Used by user_script.py for vision and embedding ONNX export.
+# -------------------------------------------------------------------------
+# Copyright (C) 2026 Advanced Micro Devices, Inc. All rights reserved.
+# Portions of this file consist of AI generated content.
+# --------------------------------------------------------------------------
+# SPDX-License-Identifier: MIT
```
[the remark this answers is not in the document]
Comment (tanzeel-amd) [reply Qwen-Qwen3.5-35B-A3B/builtin/codes/modeling_qwen3_5_moe.py]: Fixed -- SPDX identifier on line 17 now reads Apache-2.0

**Comment by VishalX** [comment]
> please wait for https://github.com/microsoft/onnxruntime-genai/pull/2146

OGA PR merged.

**Review by xieofxie — DISMISSED** [review]

**Comment by devang-ml** [comment]
Please add README and info.yml

**Comment by tanzeel-amd** [comment]
@devang-ml  Added README.md and info.yml. Please review.

**Comment by hanbitmyths** [comment]
@tanzeel-amd, we quantized embedding and vision models in qwen3.5-9b model (https://github.com/microsoft/olive-recipes/tree/main/Qwen-Qwen3.5-9B/builtin/cpu_and_mobile).

Did you see any significant quality drop after quantization of embedding and vision models? Or is it just a starting recipe before further quantization?

**Comment by VishalX** [comment]
> @tanzeel-amd, we quantized embedding and vision models in qwen3.5-9b model (https://github.com/microsoft/olive-recipes/tree/main/Qwen-Qwen3.5-9B/builtin/cpu_and_mobile).
> 
> Did you see any significant quality drop after quantization of embedding and vision models? Or is it just a starting recipe before further quantization?

I don't think we tried with quantized embedding/vision models yet, should not impact the accuracy too much.

**Review by xieofxie — APPROVED** [review]
